File: mint-runtime-permission/src/main/java/net/crystalapps/permission/runtime/library/utils/PersistentUtil.java
package net.crystalapps.permission.runtime.library.utils;

import android.content.Context;
import android.content.SharedPreferences;
import android.support.annotation.NonNull;

/**
 * Created by Syed Owais Ali on 10/13/2018.
 */
public class PersistentUtil {

    public static boolean isCameFirst(@NonNull Context context, @NonNull String permType) {
        SharedPreferences sharedPreferences = context.getSharedPreferences("mint_core_rp", Context.MODE_PRIVATE);
        boolean isFirstTime = sharedPreferences.getBoolean(permType, true);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putBoolean(permType, false);
        editor.apply();
        return isFirstTime;
    }
}

File: mint-runtime-permission/src/main/java/net/crystalapps/permission/runtime/library/utils/PersistentUtil.kt
package net.crystalapps.permission.runtime.library.utils

import android.content.Context
import android.content.SharedPreferences

/**
 * Created by Syed Owais Ali on 10/13/2018.
 */
object PersistentUtil {

    fun isCameFirst(context: Context, permType: String): Boolean {
        val sharedPreferences = context.getSharedPreferences("mint_core_rp", Context.MODE_PRIVATE)
        val isFirstTime = sharedPreferences.getBoolean(permType, true)
        val editor = sharedPreferences.edit()
        editor.putBoolean(permType, false)
        editor.apply()
        return isFirstTime
    }
}
